File: blog-backend/blog-server/src/main/java/com/yvmoux/blog/entity/Comment.java
package com.yvmoux.blog.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("comment")
public class Comment {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long articleId;

    private Long userId;

    private String content;

    private Long parentId;

    private String status;

    private Integer isRead;

    private LocalDateTime createdAt;
}

File: front/packages/shared/src/types/comment.ts
import type { User } from './user'

export interface Comment {
  id: number
  content: string
  user: {
    id: number
    username: string
    avatar?: string | null
  }
  replyTo?: {
    id: number
    username: string
  } | null
  createdAt: string
}

export interface ReplyNotification {
  id: number
  content: string
  articleId: number
  articleTitle: string
  user: {
    id: number
    username: string
  }
  isRead: boolean
  createdAt: string
}

export interface CommentFormData {
  content: string
  parentId?: number | null
}
